src/wdio/services/SeleniumDockerService.js

```diff
@@ -0,0 +1,121 @@
+import { exec } from 'child_process';
+import retry from 'async/retry';
+import http from 'http';
+import path from 'path';
+import fs from 'fs';
+
+/**
+* Webdriver.io SeleniuMDockerService
+* provides standalone chrome/firefox selenium docker automation.
+*/
+export default class SeleniumDockerService {
+  constructor() {
+    this.getSeleniumStatus = this.getSeleniumStatus.bind(this);
+  }
+
+  getSeleniumStatus(callback) {
+    http.get({
+      host: this.host,
+      port: this.port,
+      path: path.join(this.path || '/wd/hub', 'status'),
+    }, (res) => {
+      const { statusCode } = res;
+      if (statusCode !== 200) {
+        callback('Request failed');
+        return;
+      }
+
+      res.setEncoding('utf8');
+      let rawData = '';
+      res.on('data', (chunk) => { rawData += chunk; });
+      res.on('end', () => {
+        try {
+          const status = JSON.parse(rawData);
+          if (status.value && status.value.ready) {
+            callback(null, status);
+          } else {
+            callback(status);
+          }
+        } catch (e) {
+          callback(`Request failed: ${e.message}`);
+        }
+      });
+    }).on('error', (e) => {
+      callback(`Request failed: ${e.message}`);
+    });
+  }
+
+  /**
+   * Start up docker container before all workers get launched.
+   * @param {Object} config wdio configuration object
+   * @param {Array.<Object>} capabilities list of capabilities details
+   */
+  onPrepare(config, capabilities) {
+    this.config = {
+      cidfile: '.docker_selenium_id', // The docker container id file
+      enabled: true, // True if service enabled, false otherwise
+      cleanup: false, // True if docker container should be removed after test
+      image: null, // The image name to use, defaults to selenium/standalone-${browser}
+      ...(config.seleniumDocker || {}),
+    };
+    this.host = config.host;
```

Comment:
In the example provided in the README, consumer's specif the browserURL in the config, where is this passed here? Also, is host or path ever set? It defaults weren't provided in `src/wdio/conf.js`

Reply:
host is the selenium host which is different than the browserURL. I can't really default setup the browser URL in the base config being it depends on external factors (is IP remote or local, what is port).